### DISCUSSION TITLE
WebRTC: Meaning of "initiator ICE candidate error"

### DISCUSSION THREAD

**Discussion by 2color** [libp2p/js-libp2p#2530] [Q&A]
In the universal-connectivity, we [initialise the webrtc transport in js-libp2p as follows](https://github.com/libp2p/universal-connectivity/blob/2246b251a1a8cf160a9131279aa8c5198d3e0880/js-peer/src/lib/libp2p.ts#L47-L56):
```js
      webRTC({
        rtcConfiguration: {
          iceServers: [
            {
              // STUN servers help the browser discover its own public IPs
              urls: ['stun:stun.l.google.com:19302', 'stun:global.stun.twilio.com:3478'],
            },
          ],
        },
      }),
```

When I initiate a WebRTC connection from the browser to another browser peer with the following multiaddr: `/ip4/147.28.186.157/udp/9090/webrtc-direct/certhash/uEiBbC9bbdvraVWDvcvCEdJAWDymmUqiJQ964FuyEq0hELw/p2p/12D3KooWGahRw3ZnM4gAyd9FK75v4Bp5keFYTvkcAwhpEm28wbV3/p2p-circuit/webrtc/p2p/12D3KooWNuYo4UdN3eomAngE4DPRwVz5qXxLQsa6njw1TjFJZsNb`, I noticed that I get the following error:

![Screenshot 2024-05-07 at 11 58 28 am](https://github.com/libp2p/js-libp2p/assets/1992255/83cb1d18-a935-4445-bd59-ee11957b6a2c)

What's the exact meaning of this error? 

What are some good ways to get a better understanding and debug the state of a webrtc connection? 

I should note that this is a connection between two browsers on the same machine and the connection is upgraded successfully to a direct connection.

**Comment by achingbrain** [comment]
I think it means your browser is failing to resolve the domain name of the configured servers.  There's an `errorText` field in there:

```
STUN host lookup received error
```

I don't know if including the `stun:` protocol at the start of the url string might be tripping it up?  Just a guess.

- https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/icecandidateerror_event

**Reply by 2color** [reply]
I think including `stun:` is correct based on other examples I've seen in the wild. But since it failed with both STUN servers, I think this may have been related to the network I tried this on. 

My hunch is that it's related to IPv6 (see addresses). I'll try it again on a network with IPv6 enabled. But from what I know, IPv6 reliability depends to a large degree on the ISPs ability to route packets, which apparently not all ISPs do well.
